sql/sqlwire/wire.proto

```diff
@@ -102,6 +100,16 @@ message Request {
   repeated Datum params = 3 [(gogoproto.nullable) = false];
 }
 
+// ResponseWithError is what we send on the wire. It includes the error so that
+// Response doesn't need to, simplifying error handling that involves handling
+// these objects.
+message ResponseWithError {
```

Comment:
Eh, that's true, but the overhead of having this extra type is basically 3 lines of code. The separation makes it explicit that this type should not be used except just before going to the wire. In other words, this code could have been written in this style in the first place, and yet it wasn't.